src/build.cc

```diff
@@ -1071,8 +1074,9 @@ bool Builder::ExtractDeps(CommandRunner::Result* result,
   } else if (deps_type == "gcc" || deps_type == "msvc_source_dependencies") {
     string depfile = result->edge->GetUnescapedDepfile();
     if (depfile.empty()) {
-      *err = string("edge with deps=") + deps_type +
-             string("but no depfile makes no sense");
+      *err = string("edge with deps=");
+      err->append(deps_type);
```

Comment:
Please keep using `+`.

Reply:
```
string("edge with deps=") + deps_type + string("but no depfile makes no sense");
```

transforms into
```
string(string(string("") + deps_type) + string())
```

Which involves 4 allocates and 4 copies, of progressively larger amounts of data.

using append only involves 3 allocates of the same buffer to resize it, and 3 copies of only part of the total data.

Better would be to also use err->reserve(err->size() + deps_type + charcount_of_string_literals());, which takes it down to 1 allocation total, and 1 copies of each part of the data.

Reply:
1. No one really cares
2. Perfect solution looks ugly AF

Reply:
> No one really cares

I do. It's a performance issue.

> Perfect solution looks ugly AF

Then allow C++17, and use this: https://github.com/splinter-build/splinter/blob/10-string_concat/src/string_concat.h

Reply:
> Then

Or ... just use +.

Reply:
> Or ... just use +.

Which, again, is not CPU efficient.

Reply:
Correct. It's a trade-off.

Reply:
CPU efficiency is more important to me than compatibility with unspecified upgradable legacy platforms.

Since you care about the legacy platforms, don't complain about code that is objectively more efficient if it only requires a single line of "ugly".

Reply:
Isn't this error handling code, when would this be performance critical?

Reply:
I'm not planning to die on this hill,

but it's performance critical when error handling code is doing allocations and storing the error code in a std::string for other parts of the binary to reference later.

We can't re-architect the Ninja error handling method in this issue, but we can (and already have) avoid an obvious inefficency

Reply:
I would say it's the opposite: This is error handling code, where it isn't performance critical.

Reply:
Changed back to `+`.